clone.go

```diff
@@ -0,0 +1,18 @@
+package fsutil
+
+import (
+	"fmt"
+
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+func cloneProto[T protoreflect.ProtoMessage](from T) (T, error) {
+	var zero T
+	clone := proto.Clone(from)
```

Comment:
I'm a bit worried about including this reflect-based clone in this internal logic where no reflect should be needed. Are there other options?

Reply:
Eg. maybe some of these private functions should not use proto based types.

Reply:
Ended up just doing the below.

```
	clone := &types.Stat{
		Path:     from.Path,
		Mode:     from.Mode,
		Uid:      from.Uid,
		...
	}
```